airflow/serialization/pydantic/taskinstance.py

```diff
@@ -458,9 +458,9 @@ def schedule_downstream_tasks(self, session: Session | None = None, max_tis_per_
 
         :meta: private
         """
-        return TaskInstance._schedule_downstream_tasks(
-            ti=self, session=session, max_tis_per_query=max_tis_per_query
-        )
+        # we should not schedule downstream tasks with Pydantic model because it will not be able to
+        # get the DAG object (we do not serialize it currently).
+        return
```

Comment:
Ah, cool, with this simple trick just "nothing is scheduled, so far I explicitly disabled mini scheduler... but if no task is returned here also nothing is attempted to be scheduled...

Reply:
Yep

Reply:
you have the dag locally

i think what you would need to do is just add `dag` as param to this function, then pass it separately, then inside the "real" function do 
```
if ti.task.dag is ATTRIBUTE_REMOVED:
     ti.task.dag = dag
```
or something like that

Reply:
Yeah. I even attempted that, but the thing is that it is rather useless - effectively this one is run in a remote component (internal API) that de snot have the DAG object and the only way to get that DAG object is to parse it.  The whole idea of mini scheduler is that we already have DAG object so that we can get the downstream deps - and it would also mean that we have to parse the DAG in the internal_api component  - or serialize and parse the DAG object from the worker. I'd opt for the latter - because if we start parsing the DAG in the internal-api component, this has the potential of the DAG parsing accessing the DB directly (internal_api object has access to the DB). Not mentioning that most of the benefits of the mini-scheduler  (DAG already loaded in memory) are way less

So the easiest way is to just skip mini-scheduler for now. But we can bring it back later.

Reply:
preface: i don't care about mini scheduler, so i'm not saying you need to implement this method.
but just on the point... i don't think you need to parse the dag. the dag has already been parsed. whenever we run a task, we do so after parsing the dag again.  and we pass the dag object around in earlier parts in the task execution process.

Reply:
But the problem is that this method is a "db" method and it needs to be executed in the "internal_api" component - so we would have to effectively pass the parsed DAG over the RPC call (or parse the DAG in the "internal_api". This method however intermixes `dag` object and  DB access. This is also why I think mini-scheduler in AIP-72 is going to be very difficult, because (at least currently) you would essentially have to have both "parsed DAG" and DB in the same place (with AIP-72 and internal API here  - the worker has DAG, but it has no DB.